pages/_app.js

```diff
@@ -3,12 +3,29 @@ import Layout from "@/components/Layout/Layout";
 import initialFlashcards from "@/assets/flashcards.json";
 import collections from "@/assets/collections.json";
 import useLocalStorageState from "use-local-storage-state";
+import { uid } from "uid";
 
 export default function App({ Component, pageProps }) {
   const [flashcards, setFlashcards] = useLocalStorageState("flashcards", {
     defaultValue: initialFlashcards,
   });
 
+  function handleCreateFlashcard(newFlashcard) {
+    const currentCollection = collections.find((collection) => {
+      return collection.title === newFlashcard.collection;
+    });
+
+    setFlashcards([
+      {
+        id: uid(),
+        collectionId: currentCollection.id,
+        question: newFlashcard.question,
+        answer: newFlashcard.answer,
+      },
+      ...flashcards,
+    ]);
```

Comment:
**Optional:** You could use the spread operator to automatically include the "question" and "answer" fields. 


```suggestion
    setFlashcards([
      {
        id: uid(),
        collectionId: currentCollection.id,
        ...newFlashcard,
      },
      ...flashcards,
    ]);
```

Reply:
We didn't do it like this, because the object of newFlashcard includes other key-value pairs, too, like collection with the value of the collection name. Since we don't want to include the collection name in the flashcard itself but in the separate array like in the given data structure, we solved it like above. 

Is there a simple way to spread the newFlashcard into the array and leave one of the key-value pairs out or delete it afterwards?

Reply:
>  Since we don't want to include the collection name in the flashcard itself but in the separate array like in the given data structure, we solved it like above.

We should probably discuss this point again in tomorrow's Daily. As far as I understand, you are reconstructing the FlashCard along with the collection name within the code base. One idea might be to set the value of the select field in the form to the collection ID.